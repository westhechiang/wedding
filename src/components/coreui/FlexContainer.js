import styled from 'styled-components';
import { theme } from '../../ui';

const padding = {
  padTop: `${theme.gutter} 0 0 0`,
  padBottom: `0 0 ${theme.gutter} 0`,
  padTopBottom: `${theme.gutter} 0`,
  padLeft: `0 0 0 ${theme.gutter}`,
  padRight: `0 ${theme.gutter} 0 0`,
  padLeftRight: `0 ${theme.gutter}`,
  padLeftRightBottom: `0 ${theme.gutter} ${theme.gutter}`,
  padLeftRightTop: `${theme.gutter} ${theme.gutter} 0`,
  padAll: `${theme.gutter}`,
};

const background = (type, content) => {
  const types = {
    linear: `
      background: ${content},
    `,
    image: `
      background-image: url(${content});
      background-size: cover;
      background-repeat: no-repeat;
    `,
    undefined: 'transprent',
  };
  return types[type];
};

// FIXME: [WES] - heights should be removed and we should just pass in the exact
// height we want to use. Don't have time right now to go through our app and
// do a find and replace. #INTERNWORK
const heights = {
  fullHeight: '100%',
  fullVhMinusNav: 'calc(100vh - 246px)',
  fullHeightMinusNav: 'calc(100% - 246px)',
  fullVhMinusRefineListFooter: 'calc(100vh - 70px)',
  undefined: 'auto',
};

const width = {
  fullWidth: '100%',
  halfWidth: '50%',
  inherit: 'inherit',
  undefined: 'auto',
};

const FlexContainer = styled.div`
  -webkit-font-smoothing: antialiased;
  display: ${props => props.display || 'flex'};
  flex-wrap: ${props => props.flexWrap || 'nowrap'};
  flex-direction: ${props => props.flexDirection};
  color: ${props => theme.color[props.color] || theme.color.gray2};
  position: ${props => (props.position || 'relative')};
  align-content: ${props => props.alignContent};
  justify-content: ${props => props.justifyContent};
  align-items: ${props => props.alignItems};
  width: ${props => width[props.width] || props.width};
  padding: ${props => padding[props.padding] || props.padding};
  border-radius: ${props => (props.borderRadius || 0)};
  margin: ${props => props.margin || 0};
  box-shadow: ${props => (props.boxShadow ? theme.shadow : 'none')};
  font-size: ${props => theme.font.size[props.size] || theme.font.size.default};
  font-family: ${theme.font.family};
  font-weight: ${props => theme.font.weight[props.weight] || theme.font.weight.medium};
  overflow-x: ${props => props.overflowX || 'visible'};
  overflow-y: ${props => props.overflowY || 'visible'};
  box-sizing: border-box;
  top: ${props => props.top || 'auto'};
  right: ${props => props.right || 'auto'};
  left: ${props => props.left || 'auto'};
  bottom: ${props => props.bottom || 'auto'};
  z-index: ${props => props.zIndex || 'auto'};
  transform: ${props => props.transform || 'none'};
  transition: ${props => props.transition || 'all 0s 0s ease'};
  border: ${props => props.border || 'none'};
  ${props => background(props.backgroundType, props.background)}
  flex-shrink: ${props => props.flexShrink || 1};
  align-self: ${props => (props.alignSelf || 'auto')};
  height: ${props => heights[props.height] || props.height};
`;

export default FlexContainer;
