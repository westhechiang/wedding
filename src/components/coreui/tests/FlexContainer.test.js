import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FlexContainer from '../../../components/coreui/FlexContainer';

describe('<FlexContainer />', () => {
  it('should match its empty snapshot', () => {
    const wrapper = shallow(<FlexContainer />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "padTop" snapshot', () => {
    const wrapper = shallow(<FlexContainer padding="padTop" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "padBottom" snapshot', () => {
    const wrapper = shallow(<FlexContainer padding="padBottom" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "padTopBottom" snapshot', () => {
    const wrapper = shallow(<FlexContainer padding="padTopBottom" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "padLeft" snapshot', () => {
    const wrapper = shallow(<FlexContainer padding="padLeft" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "padRight" snapshot', () => {
    const wrapper = shallow(<FlexContainer padding="padRight" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "padAll" snapshot', () => {
    const wrapper = shallow(<FlexContainer padding="padAll" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "borderBottomLight" snapshot', () => {
    const wrapper = shallow(<FlexContainer border="borderBottomLight" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "borderBottom" snapshot', () => {
    const wrapper = shallow(<FlexContainer border="borderBottom" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "borderTop" snapshot', () => {
    const wrapper = shallow(<FlexContainer border="borderTop" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "borderTopBottom" snapshot', () => {
    const wrapper = shallow(<FlexContainer border="borderTopBottom" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "background" snapshot', () => {
    const wrapper = shallow(<FlexContainer background="/assets/test.png" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "fullHeight" snapshot', () => {
    const wrapper = shallow(<FlexContainer height="fullHeight" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "fullVhMinusNav" snapshot', () => {
    const wrapper = shallow(<FlexContainer height="fullVhMinusNav" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "fullHeightMinusNav" snapshot', () => {
    const wrapper = shallow(<FlexContainer height="fullHeightMinusNav" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "fullVhMinusRefineListFooter" snapshot', () => {
    const wrapper = shallow(<FlexContainer height="fullVhMinusRefineListFooter" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "fullWidth" snapshot', () => {
    const wrapper = shallow(<FlexContainer width="fullWidth" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "inherit" snapshot', () => {
    const wrapper = shallow(<FlexContainer width="inherit" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
