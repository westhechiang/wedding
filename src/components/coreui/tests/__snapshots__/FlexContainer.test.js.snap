// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlexContainer /> should match its "background" snapshot 1`] = `
<div
  className="sc-bdVaJa faWynt"
/>
`;

exports[`<FlexContainer /> should match its "borderBottom" snapshot 1`] = `
<div
  className="sc-bdVaJa bNzhoG"
/>
`;

exports[`<FlexContainer /> should match its "borderBottomLight" snapshot 1`] = `
<div
  className="sc-bdVaJa jEAfjK"
/>
`;

exports[`<FlexContainer /> should match its "borderTop" snapshot 1`] = `
<div
  className="sc-bdVaJa fBErXD"
/>
`;

exports[`<FlexContainer /> should match its "borderTopBottom" snapshot 1`] = `
<div
  className="sc-bdVaJa bMStuU"
/>
`;

exports[`<FlexContainer /> should match its "fullHeight" snapshot 1`] = `
<div
  className="sc-bdVaJa cjiCyE"
  height="fullHeight"
/>
`;

exports[`<FlexContainer /> should match its "fullHeightMinusNav" snapshot 1`] = `
<div
  className="sc-bdVaJa jaIsYG"
  height="fullHeightMinusNav"
/>
`;

exports[`<FlexContainer /> should match its "fullVhMinusNav" snapshot 1`] = `
<div
  className="sc-bdVaJa hxslny"
  height="fullVhMinusNav"
/>
`;

exports[`<FlexContainer /> should match its "fullVhMinusRefineListFooter" snapshot 1`] = `
<div
  className="sc-bdVaJa ixpYRR"
  height="fullVhMinusRefineListFooter"
/>
`;

exports[`<FlexContainer /> should match its "fullWidth" snapshot 1`] = `
<div
  className="sc-bdVaJa duKzwZ"
  width="fullWidth"
/>
`;

exports[`<FlexContainer /> should match its "inherit" snapshot 1`] = `
<div
  className="sc-bdVaJa ioGJAa"
  width="inherit"
/>
`;

exports[`<FlexContainer /> should match its "padAll" snapshot 1`] = `
<div
  className="sc-bdVaJa bdwHOc"
/>
`;

exports[`<FlexContainer /> should match its "padBottom" snapshot 1`] = `
<div
  className="sc-bdVaJa kXgZkd"
/>
`;

exports[`<FlexContainer /> should match its "padLeft" snapshot 1`] = `
<div
  className="sc-bdVaJa bhLxXO"
/>
`;

exports[`<FlexContainer /> should match its "padRight" snapshot 1`] = `
<div
  className="sc-bdVaJa dDRDob"
/>
`;

exports[`<FlexContainer /> should match its "padTop" snapshot 1`] = `
<div
  className="sc-bdVaJa ciLlrN"
/>
`;

exports[`<FlexContainer /> should match its "padTopBottom" snapshot 1`] = `
<div
  className="sc-bdVaJa eIisyv"
/>
`;

exports[`<FlexContainer /> should match its empty snapshot 1`] = `
<div
  className="sc-bdVaJa ehddzP"
/>
`;
